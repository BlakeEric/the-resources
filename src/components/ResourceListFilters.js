import React from "react"

export default function ResourceListFilters(props) {

  return (
    <form>
      <h3>Filter!</h3>

      <label htmlFor="searchTerm">
        Name
        <input
          aria-label="Name"
          type="text"
          id="searchTerm"
          name="searchTerm"
          placeholder="search resources"
          value={props.searchTerm || ""}
          onChange={(event) => {props.setSearchTerm(event.target.value)}}
        />
      </label>

      <label htmlFor="resourceSkillLevel">Skill Level</label>
      <select
        id="resourceSkillLevel"
        name="resourceSkillLevel"
        onChange={(event) => {props.setSelected('selectedSkillLevel', event.target.value)}}
        value={props.selectedSkillLevel || ""}
      >
        <option value="">All</option>
        {props.skillLevels.map((skillLevel, i) =>
          <option key={`skillLevel-${i}`} value={skillLevel}>{skillLevel}</option>)
        }
      </select>

      <label htmlFor="resourceFormat">Format</label>
      <select
        id="resourceFormat"
        name="resourceFormat"
        onChange={(event) => {props.setSelected('selectedFormat', event.target.value)}}
        value={props.selectedFormat || ""}
      >
        <option value="">All</option>
        {props.formats.map((format, i) =>
          <option key={`format-${i}`} value={format}>{format}</option>)
        }
      </select>

      <label htmlFor="resourceFocus">Focus</label>
      <select
        id="resourceFocus"
        name="resourceFocus"
        onChange={(event) => {props.setSelected('selectedFocus', event.target.value)}}
        value={props.selectedFocus || ""}
      >
        <option value="">All</option>
        {props.focuses.map((focus, i) =>
          <option key={`focus-${i}`} value={focus}>{focus}</option>)
        }
      </select>

    </form>
  )
}
