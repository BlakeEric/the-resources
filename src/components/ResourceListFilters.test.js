import React from "react"
import renderer from "react-test-renderer"
import ResourceListFilters from "./ResourceListFilters"

let mockProps = {
  selectedSkillLevel: null,
  selectedFormat: null,
  selectedFocus: null,
  setSearchTerm: () => {},
  setSelected: () => {},
  skillLevels: ["Beginner", "Moderate", "Advanced"],
  focuses: ["Front-end", "Multiple", "Algorithms"],
  formats: ['Article', 'Interactive', 'Tutorial', 'Video'],
  searchTerm: null,
}

describe("ResourceListFilters", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ResourceListFilters {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Adds selected values to appropriate fields", () => {

    mockProps = {
      ...mockProps,
      selectedSkillLevel: "Beginner",
      selectedFormat: "Article",
      selectedFocus: "Front-end",
      searchTerm: "Code"
    }
    const tree = renderer.create(<ResourceListFilters {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
