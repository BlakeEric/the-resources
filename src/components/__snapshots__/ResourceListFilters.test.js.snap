// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceListFilters Adds selected values to appropriate fields 1`] = `
<form
  css={
    Object {
      "map": undefined,
      "name": "3r6u4d",
      "next": undefined,
      "styles": "
  font-size: 0.8855rem;
  margin: auto -0.805rem;
  fieldset {
    color: white;
    border-radius: 5px;
    background-color: #0acc9e;
    background-image: linear-gradient(to bottom right, #37e6bd, #0acc9e);
    padding: 0.4025rem;
    border: none;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.05);
    legend {
      width: 100%;
      float: left;
      padding: 0 0.4025rem;
      font-size: 0.805rem;
      letter-spacing: 1px;
      line-height: 1.5;
      border-radius: 5px;
      text-transform: uppercase;
      button.btn-clearFilters {
        margin-left: 12px;
        display: inline;
        padding: 2px .25rem;
        color: #b90000;
        border: 1px solid #b90000;
        border-radius: 3px;
        background: white;
        font-size: 0.805rem;
        line-height: 1;
        &:hover, &:focus {
          color: #800000;
          border: 1px solid #800000;
        }
      }
    }
  }
  label {
    width: 50%;
    display: inline-block;
    padding: 0.4025rem;
    @media (min-width: 600px) {
      width: 25%;
    }
  }
  input, select {
    width: 100%;
    height: 30px;
    border-color: white;
    box-shadow: none;
    border-radius: 5px;
    padding-left: 5px;
    padding-right: 5px;
    border: 1.5px solid white;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.05);
  }
",
      "toString": [Function],
    }
  }
>
  <fieldset>
    <legend>
      Filter
      <button
        className="btn-clearFilters"
      >
        Ã— Clear All
      </button>
    </legend>
    <label
      htmlFor="searchTerm"
    >
      Name
      <input
        aria-label="Name"
        id="searchTerm"
        name="searchTerm"
        onChange={[Function]}
        placeholder="Start typing..."
        type="text"
        value="Code"
      />
    </label>
    <label
      htmlFor="resourceFormat"
    >
      Format
      <select
        id="resourceFormat"
        name="resourceFormat"
        onChange={[Function]}
        value="Article"
      >
        <option
          value=""
        >
          All
        </option>
        <option
          value="Article"
        >
          Article
        </option>
        <option
          value="Interactive"
        >
          Interactive
        </option>
        <option
          value="Tutorial"
        >
          Tutorial
        </option>
        <option
          value="Video"
        >
          Video
        </option>
      </select>
    </label>
    <label
      htmlFor="resourceFocus"
    >
      Focus
      <select
        id="resourceFocus"
        name="resourceFocus"
        onChange={[Function]}
        value="Front-end"
      >
        <option
          value=""
        >
          All
        </option>
        <option
          value="Front-end"
        >
          Front-end
        </option>
        <option
          value="Multiple"
        >
          Multiple
        </option>
        <option
          value="Algorithms"
        >
          Algorithms
        </option>
      </select>
    </label>
    <label
      htmlFor="resourceSkillLevel"
    >
      Skill Level
      <select
        id="resourceSkillLevel"
        name="resourceSkillLevel"
        onChange={[Function]}
        value="Beginner"
      >
        <option
          value=""
        >
          All
        </option>
        <option
          value="Beginner"
        >
          Beginner
        </option>
        <option
          value="Moderate"
        >
          Moderate
        </option>
        <option
          value="Advanced"
        >
          Advanced
        </option>
      </select>
    </label>
  </fieldset>
</form>
`;

exports[`ResourceListFilters renders correctly 1`] = `
<form
  css={
    Object {
      "map": undefined,
      "name": "3r6u4d",
      "next": undefined,
      "styles": "
  font-size: 0.8855rem;
  margin: auto -0.805rem;
  fieldset {
    color: white;
    border-radius: 5px;
    background-color: #0acc9e;
    background-image: linear-gradient(to bottom right, #37e6bd, #0acc9e);
    padding: 0.4025rem;
    border: none;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.05);
    legend {
      width: 100%;
      float: left;
      padding: 0 0.4025rem;
      font-size: 0.805rem;
      letter-spacing: 1px;
      line-height: 1.5;
      border-radius: 5px;
      text-transform: uppercase;
      button.btn-clearFilters {
        margin-left: 12px;
        display: inline;
        padding: 2px .25rem;
        color: #b90000;
        border: 1px solid #b90000;
        border-radius: 3px;
        background: white;
        font-size: 0.805rem;
        line-height: 1;
        &:hover, &:focus {
          color: #800000;
          border: 1px solid #800000;
        }
      }
    }
  }
  label {
    width: 50%;
    display: inline-block;
    padding: 0.4025rem;
    @media (min-width: 600px) {
      width: 25%;
    }
  }
  input, select {
    width: 100%;
    height: 30px;
    border-color: white;
    box-shadow: none;
    border-radius: 5px;
    padding-left: 5px;
    padding-right: 5px;
    border: 1.5px solid white;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.05);
  }
",
      "toString": [Function],
    }
  }
>
  <fieldset>
    <legend>
      Filter
    </legend>
    <label
      htmlFor="searchTerm"
    >
      Name
      <input
        aria-label="Name"
        id="searchTerm"
        name="searchTerm"
        onChange={[Function]}
        placeholder="Start typing..."
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="resourceFormat"
    >
      Format
      <select
        id="resourceFormat"
        name="resourceFormat"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          All
        </option>
        <option
          value="Article"
        >
          Article
        </option>
        <option
          value="Interactive"
        >
          Interactive
        </option>
        <option
          value="Tutorial"
        >
          Tutorial
        </option>
        <option
          value="Video"
        >
          Video
        </option>
      </select>
    </label>
    <label
      htmlFor="resourceFocus"
    >
      Focus
      <select
        id="resourceFocus"
        name="resourceFocus"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          All
        </option>
        <option
          value="Front-end"
        >
          Front-end
        </option>
        <option
          value="Multiple"
        >
          Multiple
        </option>
        <option
          value="Algorithms"
        >
          Algorithms
        </option>
      </select>
    </label>
    <label
      htmlFor="resourceSkillLevel"
    >
      Skill Level
      <select
        id="resourceSkillLevel"
        name="resourceSkillLevel"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          All
        </option>
        <option
          value="Beginner"
        >
          Beginner
        </option>
        <option
          value="Moderate"
        >
          Moderate
        </option>
        <option
          value="Advanced"
        >
          Advanced
        </option>
      </select>
    </label>
  </fieldset>
</form>
`;
